cardano-node/src/Cardano/Common/LocalSocket.hs

```diff
@@ -23,3 +24,7 @@ localSocketAddrInfo socketPath =
       Socket.defaultProtocol
       (Socket.SockAddrUnix socketPath)
       Nothing
+
+eitherThrow :: Exception e => (b -> e) -> Either b a ->  IO a
```

Comment:
@deepfire I'm wondering if to put this in `cardano-prelude` or to make this module more generic.